gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/util/AttributeMetadataUtil.java

```diff
@@ -55,8 +57,34 @@ public static String getTimestampAttributeId(
   }
 
   private static String getStartTimeAttributeKeyName(String attributeScope) {
-    String timestamp = TimestampConfigs.getTimestampColumn(attributeScope);
-    return timestamp == null ? START_TIME_ATTRIBUTE_KEY : timestamp;
+    return Optional.ofNullable(TimestampConfigs.getTimestampColumn(attributeScope))
+        .orElse(START_TIME_ATTRIBUTE_KEY);
+  }
+
+
+  public static String getSpaceAttributeId(
```

Comment:
Since this approach requires us to do code changes for every scope, what are the alternative ways of doing it? Probably reserved attributes across scopes maybe?

Reply:
It does generally always have the same name, with the exception of interactions which contains two variants. Most scopes don't need to be added here, just new record types (since all aggregate types use the underlying record type's space attribute.

That said, there's certainly room for improvement. An obvious one is to move this out of code and into config. A more elaborate one is to register more metadata in the attribute service (or entity service, potentially).

Reply:
Going to leave this one for later iteration unless you want me to update now.